### ISSUE TITLE
Complex regex expression results in error in the latest version (3.2.1-20200518)

### ISSUE THREAD

**Issue by suyash192** [ledger/ledger#1918]
I use the following regex expression in an automated posting to match if the transaction month is Feb, Jun or Oct.

`= /Expenses:Subscriptions:Netflix/ & expr date =~ /\/(02|06|10)\//`

This results in an error in ledger reporting commands which was not occurring prior to this version:

**Steps to reproduce:**

1. Install latest ledger version (3.2.1-20200518)
2. Add an automated posting with a complex regex which includes capturing groups.
3. Run any ledger reporting command such as balance.

**Result:**
```
$> ledger -f /path/to/file/2020.ledger bal
While parsing file "/path/to/file/2020.ledger", line 9:
While parsing value expression:
  date =~ /\/(02|06|10

While parsing automated transaction:
> = /Expenses:Subscriptions:Netflix/ & expr date =~ /\/(02|06|10)\//
Error: Missing '/'
```

**Comment by tbm** [comment]
Did this work with 3.1.x?

**Comment by tbm** [comment]
Doesn't seem to work with 3.1.x.

This briefly worked with 3.2.0 but the fix caused other regressions and was reverted in 3.2.1.

There's a ticket to track parser improvements.

**Comment by tbm** [comment]
I can confirm this briefly worked with 3.2.0 but not with 3.1.2 or 3.2.1.

This is due to John's fix which got reverted in 3.2.1 because it broke other things.

**Comment by suyash192** [comment]
Yes, this does not work in prior versions to 3.2.0 as well. I luckily just happened to start using this feature on 3.2.0 and hence never noticed before.

Would there be any temporary workout for the time being until it gets properly fixed?

**Comment by tbm** [comment]
As a workaround, you can locally apply the patch from 49b07a1c19489547b.  This made it work (for you) but broke things (for other people).

@jwiegley do you have some time this weekend to looking at the various parser issues? I tagged some with the label "parser".

**Comment by jwiegley** [comment]
Let me get taxes done, and then maybe I can take advantage of the fact that I'm on vacation for the first half of July.

**Comment by suyash192** [comment]
Just wanted an update, has there been any progress on this?

**Comment by jwiegley** [comment]
@suyash192 Hi Suyash, no progress yet, since this issue will take some doing (as in, rewriting the report query parser), and life sort of intervened since the time we first discussed this issue. Let me see if I can't get some AI assistance going to help me accelerate addressing issues like this one.

**Comment by suyash192** [comment]
> [@suyash192](https://github.com/suyash192) Hi Suyash, no progress yet, since this issue will take some doing (as in, rewriting the report query parser), and life sort of intervened since the time we first discussed this issue. Let me see if I can't get some AI assistance going to help me accelerate addressing issues like this one.

Understandable. Thanks for the update.
